fluent-syntax/src/serializer.js

```diff
@@ -23,11 +24,13 @@ export default class FluentSerializer {
       parts.push(
         `${serializeComment(resource.comment)}\n\n`
       );
+      this.hasEntries = true;
     }
 
     for (const entry of resource.body) {
       if (entry.types !== 'Junk' || this.withJunk) {
         parts.push(this.serializeEntry(entry));
+        this.hasEntries = true;
```

Comment:
I would avoid setting it in every pass of the loop and only do it if `hasEntries` is `false`.